af/fastpath/fastpath/tests/test_functional.py

```diff
@@ -1,40 +1,74 @@
 """
 Simulate feeding from the collectors or cans on S3 using a local can
 """
+# Format with black -t py37 -l 120
 
+from datetime import date, timedelta
+from collections import Counter
 from pathlib import Path
-import hashlib
 import logging
 import os
-import time
 
 import ujson
 import boto3  # debdeps: python3-boto3
 import pytest  # debdeps: python3-pytest
 
 import fastpath.core as fp
-
-# from fastpath.utils import fingerprints
+import fastpath.s3feeder as s3feeder
 
 log = logging.getLogger()
 
+# The fixtures download cans from S3 to a local directory
+#
+# Use credentials from ~/.aws/config in the block:
+# [ooni-data-private]
+# aws_access_key_id = ...
+# aws_secret_access_key = ...
+#
+# Explore bucket from CLI:
+# AWS_PROFILE=ooni-data-private aws s3 ls s3://ooni-data-private/canned/2019-07-16/
 
-@pytest.fixture
-def cans():
-    """
-    Download interesting cans from S3 to a local directory
+BUCKET_NAME = "ooni-data-private"
```

Comment:
It's unfortunate we can't run these on the CI due to these files being in the private bucket. We should evaluate if we should open up the cans and move them to the public s3 bucket.

Reply:
As discussed before, the fastpath is publishing the msmts in real time on purpose it publishing the raw cans should be the same. However we could also have a dedicated s3 user configured in Travis CI